lib/db/schema/patch-045-046.sql

```diff
@@ -0,0 +1,188 @@
+-- Add emails table and corresponding procedures to
+-- create, get and delete them.
+
+CREATE TABLE IF NOT EXISTS emails (
+  normalizedEmail VARCHAR(255) NOT NULL PRIMARY KEY,
+  email VARCHAR(255) NOT NULL,
+  uid BINARY(16) NOT NULL,
+  emailCode BINARY(16) NOT NULL,
+  isVerified BOOLEAN NOT NULL DEFAULT FALSE,
+  isPrimary BOOLEAN NOT NULL DEFAULT FALSE,
+  verifiedAt BIGINT UNSIGNED,
+  createdAt BIGINT UNSIGNED NOT NULL
+) ENGINE=InnoDB;
+
+DROP procedure IF EXISTS `createEmail_1`;
+
+CREATE PROCEDURE `createEmail_1` (
+    IN `normalizedEmail` VARCHAR(255),
+    IN `email` VARCHAR(255),
+    IN `uid` BINARY(16) ,
+    IN `emailCode` BINARY(16),
+    IN `isVerified` TINYINT(1),
+    IN `isPrimary` TINYINT(1),
+    IN `verifiedAt` BIGINT UNSIGNED,
+    IN `createdAt` BIGINT UNSIGNED
+)
+BEGIN
+    DECLARE EXIT HANDLER FOR SQLEXCEPTION
+        BEGIN
+        ROLLBACK;
+        RESIGNAL;
+    END;
+
+    START TRANSACTION;
+
+    -- Currently, can not add an email that is specified in the
+    -- accounts table, regardless of verification state.
```

Comment:
> Do you mean inserting email in both tables?

Yes.  It might turn out *not* to be cleaner overall, but maybe worth a look:

* When you first create the account, write the `email` field on the `accounts` table *and* insert a row into the `emails` table with `isPrimary=true`.
* Somehow backfill the primary emails for existing accounts into this table; this will probably have to be on-demand as you describe above, whenever you discover that the user has no emails in the `emails` table, add their primary email in.

Hmm...now that I try to describe it in more words, that "somehow backfill" hand-wavery sounds like it might be more complicated than it's worth.  I'll certainly defer to your judgement on which way is simpler overall!